invenio_pidrelations/serializers/schemas.py

```diff
@@ -61,37 +62,38 @@ def _dump_relative(self, relative):
 
     def dump_next(self, obj):
         """Dump the parent of a PID."""
-        if self._is_child(obj):
-            return self._dump_relative(obj.next)
+        if self._is_child(obj) and not obj.is_last_child(self.context['pid']):
+            return self._dump_relative(obj.next_child(self.context['pid']))
 
     def dump_previous(self, obj):
         """Dump the parent of a PID."""
-        if self._is_child(obj):
-            return self._dump_relative(obj.previous)
+        if self._is_child(obj) and obj.index(self.context['pid']) > 0:
+            return self._dump_relative(obj.previous_child(self.context['pid']))
 
     def dump_index(self, obj):
         """Dump the index of the child in the relation."""
-        if obj.is_ordered and self._is_child(obj):
-            return obj.index
+        if isinstance(obj, PIDNodeOrdered) and self._is_child(obj):
+            return obj.index(self.context['pid'])
         else:
             return None
 
     def _is_parent(self, obj):
         """Check if the PID from the context is the parent in the relation."""
-        return obj.parent == self.context['pid']
+        return self.context['pid'] == obj.pid
```

Comment:
`and obj.is_parent()` ?

Reply:
Documentation will explain that `dump` should only be called with `obj` being a parent NodePID.

Done